refactor: run sourcery on all files

barredterra opened the pull request:
<details>
<summary>Issues by count</summary>

```
┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━┓
┃ Rule ID                               ┃ Count ┃
┡━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━┩
│ avoid-using-var                       │  1155 │
│ use-fstring-for-concatenation         │   482 │
│ dont-reassign-parameters              │   201 │
│ no-loop-in-tests                      │   183 │
│ avoid-function-declarations-in-blocks │   160 │
│ inline-immediately-returned-variable  │   157 │
│ assign-if-exp                         │   141 │
│ no-conditionals-in-tests              │   111 │
│ simplify-ternary                      │    71 │
│ de-morgan                             │    69 │
│ dont-import-test-modules              │    68 │
│ list-comprehension                    │    60 │
│ remove-unnecessary-else               │    52 │
│ swap-if-else-branches                 │    50 │
│ low-code-quality                      │    49 │
│ reintroduce-else                      │    49 │
│ merge-else-if                         │    42 │
│ swap-if-expression                    │    36 │
│ boolean-if-exp-identity               │    35 │
│ swap-nested-ifs                       │    35 │
│ for-append-to-extend                  │    32 │
│ replace-interpolation-with-fstring    │    30 │
│ raise-specific-error                  │    29 │
│ simplify-dictionary-update            │    29 │
│ merge-else-if-into-elif               │    28 │
│ dict-assign-update-to-union           │    27 │
│ or-if-exp-identity                    │    25 │
│ remove-redundant-if                   │    25 │
│ assignment-operator                   │    23 │
│ use-or-for-fallback                   │    23 │
│ move-assign-in-block                  │    22 │
│ introduce-default-else                │    20 │
│ invert-ternary                        │    20 │
│ switch                                │    18 │
│ move-assign                           │    17 │
│ last-if-guard                         │    16 │
│ merge-duplicate-blocks                │    16 │
│ dict-comprehension                    │    14 │
│ raise-from-previous-error             │    13 │
│ hoist-similar-statement-from-if       │    10 │
│ flip-comparison                       │     9 │
│ lift-return-into-if                   │     9 │
│ dict-literal                          │     8 │
│ hoist-statement-from-if               │     8 │
│ merge-list-appends-into-extend        │     8 │
│ remove-pass-body                      │     8 │
│ remove-redundant-pass                 │     8 │
│ split-or-ifs                          │     8 │
│ comprehension-to-generator            │     7 │
│ identity-comprehension                │     7 │
│ invert-any-all                        │     7 │
│ merge-list-append                     │     7 │
│ no-eval                               │     7 │
│ no-new-function                       │     7 │
│ use-any                               │     7 │
│ collection-into-set                   │     6 │
│ max-min-identity                      │     6 │
│ merge-comparisons                     │     6 │
│ merge-dict-assign                     │     6 │
│ use-getitem-for-re-match-groups       │     6 │
│ remove-str-from-fstring               │     5 │
│ use-join                              │     5 │
│ avoid-builtin-shadow                  │     4 │
│ remove-redundant-slice-index          │     4 │
│ remove-zero-from-range                │     4 │
│ simplify-len-comparison               │     4 │
│ aug-assign                            │     3 │
│ aware-datetime-for-utc                │     3 │
│ class-method-first-arg-name           │     3 │
│ hoist-statement-from-loop             │     3 │
│ merge-except-handler                  │     3 │
│ remove-none-from-default-get          │     3 │
│ remove-redundant-fstring              │     3 │
│ remove-unused-enumerate               │     3 │
│ simplify-generator                    │     3 │
│ use-contextlib-suppress               │     3 │
│ use-dictionary-union                  │     3 │
│ ensure-file-closed                    │     2 │
│ list-literal                          │     2 │
│ remove-redundant-continue             │     2 │
│ ternary-to-if-expression              │     2 │
│ use-datetime-now-not-today            │     2 │
│ use-dict-items                        │     2 │
│ collection-builtin-to-comprehension   │     1 │
│ default-get                           │     1 │
│ default-mutable-arg                   │     1 │
│ hoist-if-from-if                      │     1 │
│ inline-variable                       │     1 │
│ instance-method-first-arg-name        │     1 │
│ lift-duplicated-conditional           │     1 │
│ merge-isinstance                      │     1 │
│ none-compare                          │     1 │
│ possible-incorrect-bitwise-operator   │     1 │
│ remove-duplicate-dict-key             │     1 │
│ remove-empty-nested-block             │     1 │
│ remove-unit-step-from-range           │     1 │
│ remove-unreachable-code               │     1 │
│ set-comprehension                     │     1 │
│ simplify-boolean-comparison           │     1 │
│ simplify-division                     │     1 │
│ simplify-numeric-comparison           │     1 │
│ skip-sorted-list-construction         │     1 │
│ use-array-literal                     │     1 │
│ use-assigned-variable                 │     1 │
│ while-guard-to-condition              │     1 │
├───────────────────────────────────────┼───────┤
│ Total                                 │  3881 │
└───────────────────────────────────────┴───────┘
```

</details>

<details>
<summary>Sourcery config</summary>

```yaml
ignore:
  - node_modules
  - frappe/public/dist
  - frappe/public/js/lib

rule_settings:
  enable: [default]
  rule_types:
    - refactoring
  disable:
    - use-named-expression # gets kinda ugly for big expressions
    - use-next # less readable, IMO
    - dict-assign-update-to-union # less readable, IMO
    - use-braces # common pattern in this codebase
    - use-object-destructuring # less readable, IMO
    - assignment-operator # doesn't do much
    - for-index-underscore # we use underscore for the translate function
    - merge-nested-ifs # produces wrong indentation
    - remove-unnecessary-cast # produces non-equivalent refactorings
    - use-fstring-for-formatting # produces non-equivalent refactorings
    - use-fstring-for-concatenation # produces non-equivalent refactorings
    - raise-from-previous-error # produces non-equivalent refactorings
    - move-assign-in-block # suspected to produce non-equivalent refactorings
  python_version: '3.10'
```

</details>

<details>
<summary>Whitelist of useful rules with frequent issues</summary>

```
inline-immediately-returned-variable
list-comprehension
for-append-to-extend
merge-list-appends-into-extend
comprehension-to-generator
use-any
invert-any-all
merge-list-append
collection-into-set

simplify-ternary
invert-ternary
de-morgan
swap-if-else-branches
reintroduce-else
merge-else-if
swap-if-expression
boolean-if-exp-identity
swap-nested-ifs
merge-else-if-into-elif
or-if-exp-identity
remove-redundant-if
use-or-for-fallback
introduce-default-else
last-if-guard
merge-duplicate-blocks
lift-return-into-if
hoist-similar-statement-from-if
hoist-statement-from-if
merge-comparisons

replace-interpolation-with-fstring

simplify-dictionary-update
merge-dict-assign
dict-comprehension
```

</details>

ankush commented on frappe/__init__.py:
```diff
@@ -129,7 +129,7 @@ def as_unicode(text: str, encoding: str = "utf-8") -> str:
 	elif isinstance(text, bytes):
 		return str(text, encoding)
 	else:
-		return str(text)
```
This is definitely not equivalent. Few more I spotted too. Let's run rule by rule if possible with separate commits.

ankush replied:
I am guessing it's using incorrect type annotations to reach to conclusion that it's not required.

ankush commented on frappe/core/doctype/rq_job/rq_job.py:
```diff
@@ -63,8 +63,8 @@ class RQJob(Document):
 			job = Job.fetch(self.name, connection=get_redis_conn())
-		except NoSuchJobError:
-			raise frappe.DoesNotExistError
+		except NoSuchJobError as e:
```
This isn't equivalent either.

gavindsouza commented on frappe/auth.py:
```diff
@@ -405,18 +405,18 @@ def validate_ip_address(user):
-		else frappe.db.get_value("User", user, user_fields, as_dict=True)
+		frappe.db.get_value("User", user, user_fields, as_dict=True)
+		if frappe.flags.in_test
+		else frappe.get_cached_value("User", user, user_fields, as_dict=True)
```
Yikes, we should get rid of these test flags in the application source

gavindsouza commented on frappe/boot.py:
```diff
@@ -151,13 +151,11 @@ def get_allowed_report_names(cache=False) -> set[str]:
 def get_user_pages_or_reports(parent, cache=False):
 	if cache:
-		has_role = frappe.cache.get_value("has_role:" + parent, user=frappe.session.user)
+		has_role = frappe.cache.get_value(f"has_role:{parent}", user=frappe.session.user)
```
Conversions like these are scary if tests don't cover all use cases. If parent passed is None, an error would've been raised which might've been handled somewhere up the stack or at least it's a solid indicator that something is breaking. Whereas, changing this to an f-string will just evaluate it to "has_role:None"

barredterra committed:
refactor: inline immediately returned variable

Inline a variable to a return in the case when the
variable being declared is immediately returned

barredterra commented:
Well, this is more manual work than I expected. Let's stick with these two refactoring commits for now, if they're okay.

Two common patterns:

```python
x = "hello world"
return x
```

refactored to 

```python
return "hello world"
```

and

```python
out = []
for x in range(10):
	out.append(x ** x)

return out
```

refactored to 

```python
return [x ** x for x in range(10)]
```

barredterra committed:
refactor: for append to extend, merge list extend

Replace a for append loop with list extend.
Create the list with values instead of creating
an empty list and extending it with another list.

ankush commented:
*sometimes* immediately returned variables are useful for documenting what it is. Most of these cases are useless though with variable names like `result`, `out`.

ankush reviewed (approved):
Trusting sourcery and you :crossed_fingers:
